apps/test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx

```diff
@@ -0,0 +1,341 @@
+      .mockResolvedValue([
+        {nces_id: initialState.schoolId, name: initialState.schoolName},
+      ]);
+
+    sendAnalyticsEventSpy = jest.spyOn(
+      require('@cdo/apps/schoolInfo/utils/sendAnalyticsEvent'),
+      'sendAnalyticsEvent'
+    );
+  });
+
+  it('should use initialState instead of sessionStorage if passed', async () => {
+    await act(async () => {
+      sessionStorage.setItem(SCHOOL_COUNTRY, US_COUNTRY_CODE);
+      sessionStorage.setItem(SCHOOL_ID_SESSION_KEY, '2');
+      sessionStorage.setItem(SCHOOL_NAME_SESSION_KEY, 'Stored School');
+      sessionStorage.setItem(SCHOOL_ZIP_SESSION_KEY, '54321');
+
+      const {result, waitForNextUpdate} = renderHook(() =>
+        useSchoolInfo(initialState)
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current.country).toBe(initialState.country);
+      expect(result.current.schoolId).toBe(initialState.schoolId);
+      expect(result.current.schoolName).toBe(initialState.schoolName);
+      expect(result.current.schoolZip).toBe(initialState.schoolZip);
+    });
+  });
+
+  it('should use sessionStorage if no initialState is passed', async () => {
+    await act(async () => {
+      sessionStorage.setItem(SCHOOL_COUNTRY, US_COUNTRY_CODE);
+      sessionStorage.setItem(SCHOOL_ID_SESSION_KEY, '2');
+      sessionStorage.setItem(SCHOOL_NAME_SESSION_KEY, 'Stored School');
+      sessionStorage.setItem(SCHOOL_ZIP_SESSION_KEY, '54321');
+
+      const {result, waitForNextUpdate} = renderHook(() => useSchoolInfo({}));
+
+      await waitForNextUpdate();
+
+      expect(result.current.country).toBe(US_COUNTRY_CODE);
+      expect(result.current.schoolId).toBe('2');
+      expect(result.current.schoolName).toBe('Stored School');
+      expect(result.current.schoolZip).toBe('54321');
+    });
+  });
+
+  describe('hook state updates', () => {
+    let hook;
+    beforeEach(async () => {
+      await act(async () => {
+        const {result, waitForNextUpdate} = renderHook(() =>
+          useSchoolInfo(initialState)
+        );
+
+        hook = result;
+
+        await waitForNextUpdate(); // Wait for initial render
+      });
+    });
+
+    describe('country updates', () => {
+      it('should update sessionStorage', async () => {
+        act(() => {
+          hook.current.setCountry('US');
+        });
+
+        waitFor(() => {
+          const schoolCountrySessionStorageCalls =
+            mockSessionStorage.setItem.mock.calls.filter(
+              ([key]) => key === SCHOOL_COUNTRY
+            );
+          expect(schoolCountrySessionStorageCalls).toHaveLength(2);
+          expect(schoolCountrySessionStorageCalls[0][1]).toBe(
+            initialState.country
+          );
+          expect(schoolCountrySessionStorageCalls[1][1]).toBe('US');
+        });
+      });
+
+      it('should reset schoolId, schoolZip, schoolName, and schoolsList state', async () => {
+        waitFor(() => {
+          expect(hook.current.schoolId).toBe(initialState.schoolId);
```

Comment:
TIL about being able to directly reference a hook in react testing!

Reply:
for sure, I had to add a hook testing package! but yeah, it's a lot easier than testing the component that uses the hook